inc/saiport.h

```diff
@@ -174,13 +174,34 @@ typedef enum _sai_port_fec_mode_t
     /** No FEC */
     SAI_PORT_FEC_MODE_NONE,
 
-    /** Enable RS-FEC - 25G, 50G, 100G ports */
+    /** Enable RS-FEC - 25G, 50G, 100G ports. The specific RS-FEC mode will be automatically determined. */
     SAI_PORT_FEC_MODE_RS,
 
     /** Enable FC-FEC - 10G, 25G, 40G, 50G ports */
     SAI_PORT_FEC_MODE_FC,
 } sai_port_fec_mode_t;
 
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_FEC_MODE_EXTENDED
+ */
+typedef enum _sai_port_fec_mode_extended_t
+{
+    /** No FEC */
+    SAI_PORT_FEC_MODE_EXTENDED_NONE,
+
+    /** Enable RS-528 FEC - 25G, 50G, 100G ports */
+    SAI_PORT_FEC_MODE_EXTENDED_RS528,
+
+    /** Enable RS544-FEC - 100G PAM4, 200G ports */
+    SAI_PORT_FEC_MODE_EXTENDED_RS544,
+
+    /** Enable RS544-FEC (2x interleaved) - 100G, 200G, 400G ports */
+    SAI_PORT_FEC_MODE_EXTENDED_RS544_2X_INTERLEAVED,
```

Comment:
Maybe RS544_INTERLEAVED instead of RS544_2X_INTERLEAVED? The interleaved FEC sublayer does 2:1 FEC codeword interleaving but I wonder if it should be mentioned in specific (maybe if we expect other types like that to be developed though the latency may become an issue).

Reply:
Done.